react-sharing-state/my-app/src/Container.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { useState } from 'react';
+
+/**
+ * A container of items.
+ * One item is displayed at a time, with buttons to flip through them:
+ * - Next and Prev scroll forward and backward one item
+ * - An array of buttons scroll to a specific item
+ * TODO: The buttons don't work!
+ */
+export default function Container({ items }) {
+  const [isActive, setIsActive] = useState(0);
+
+  return (
+    <div>
+      <div>{items[isActive]}</div>
+      <div>
+        <CustomButton
+          text="Prev"
+          onShow={() =>
+            setIsActive((isActive - 1 + items.length) % items.length)
+          }
+        />
+        <Indicators
+          count={items.length}
+          onSelect={setIsActive}
+          index={isActive}
+        />
+        <CustomButton
+          text="Next"
+          onShow={() => setIsActive((isActive + 1) % items.length)}
+        />
+      </div>
+    </div>
+  );
+}
+
+/**
+ * A component that wraps a DOM button.
+ * Props:
+ *   text: The button's text
+ *
+ * TODO: Make the background color a prop, default white.
+ * TODO: When clicked, the parent needs to be notified.
+ */
+function CustomButton({ text, onShow, bgColor }) {
```

Comment:
Make 'white' the default. Not a big deal, but that makes the Next/Prev buttons styled the same as the indicator buttons. (And it's what the TODO says to do.)